crates/burn-import/src/onnx/dim_inference.rs

```diff
@@ -423,3 +424,12 @@ fn conv_transpose2d_update_outputs(node: &mut Node) {
         panic!("Only tensor input is valid");
     }
 }
+
+fn matmul_update_outputs(node: &mut Node) {
+    // NOTE: matmul only supported for float tensors (no broadcasting support)
+    if let ArgType::Tensor(tensor) = node.inputs[0].clone().ty {
+        node.outputs[0].ty = ArgType::Tensor(tensor);
+    } else {
+        panic!("Only tensor input is valid");
+    }
```

Comment:
We are supposed to support broadcasting, but we have to be explicit and call unsqueeze during codegen.

Reply:
Haha I was writing the same. See my comment below.